<?xml version="1.0" ?>
<launch>
	<!-- <arg name="load_gripper" default="true" /> -->

	<!-- Franka control -->
	<!-- <arg name="robot_ip" default="172.16.0.2"/>
	<arg name="load_gripper" default="true" />
	<include file="$(find franka_control)/launch/franka_control.launch" >
		<arg name="robot_ip" value="$(arg robot_ip)" />
		<arg name="load_gripper" value="$(arg load_gripper)" />
	</include> -->

	<!-- Set some project specific defaults for the panda arm. -->
	<!-- <node name="set_robot_defaults" pkg="franka_irom_controllers" type="set_panda_defaults.py" respawn="false" output="screen"/> -->

	<!-- Velocity controller for continuous task -->
	<!-- <include file="$(find franka_irom_controllers)/launch/cartesian_velocity_node_controller.launch">
		<arg name="stopped" value="true" />
	</include> -->

	<!-- <node name="scoop" pkg="franka_irom" type="scoop_demo_new.py" output="screen"/> -->
	<arg name="num_trial" default="4" />
	<arg name="recording_k4a" default="False" />
	<arg name="recording_rs" default="False" />
	<arg name="video_folder_prefix" default="" />
	<node name="push" pkg="franka_irom" type="push.py" output="screen" args="$(arg num_trial) $(arg recording_rs) $(arg recording_k4a) $(arg video_folder_prefix)"/>

	<node name="push_camera" pkg="franka_irom" type="push_camera_env.py" output="screen"/>

	<group if="$(arg recording_rs)">
		<node name="rs_record" pkg="franka_irom" type="rs_recorder.py" output="screen"/>
	</group>
	<group if="$(arg recording_k4a)">
		<node name="k4a_record" pkg="franka_irom" type="k4a_recorder.py" output="screen"/>
	</group>
</launch>

